core/src/Cuda/Kokkos_Cuda_Graph_Impl.hpp

```diff
@@ -64,6 +65,11 @@ struct GraphImpl<Kokkos::Cuda> {
              ->cuda_graph_instantiate_wrapper(&m_graph_exec, m_graph,
                                               &error_node, error_log,
                                               error_log_size)));
+#else
+    KOKKOS_IMPL_CUDA_SAFE_CALL(
+        (m_execution_space.impl_internal_space_instance()
+             ->cuda_graph_instantiate_wrapper(&m_graph_exec, m_graph)));
```

Comment:
Keep this and remove the rest.
Move the guards to the wrapper function implementation.